osbs/build/user_params.py

```diff
@@ -0,0 +1,183 @@
+                   reactor_config_map=None,
+                   yum_repourls=None, signing_intent=None, compose_ids=None,
+                   isolated=None, scratch=None,
+                   **kwargs):
+        self.git_uri.value = git_uri
+        self.git_ref.value = git_ref
+        self.git_branch.value = git_branch
+        self.user.value = user
+        self.component.value = component
+        self.release.value = release
+        self.build_type.value = build_type
+
+        self.name.value = make_name_from_git(self.git_uri.value, self.git_branch.value)
+        self.reactor_config_map.value = reactor_config_map
+
+        unique_build_args = (build_imagestream, build_image, build_from)
+        if sum(bool(a) for a in unique_build_args) != 1:
+            raise OsbsValidationException(
+                'Please only define one of build_from, build_image, build_imagestream')
+        self.build_image.value = build_image
+        self.build_imagestream.value = build_imagestream
+        if self.build_image.value or self.build_imagestream.value:
+            logger.warning("build_image or build_imagestream is defined, they are deprecated,"
+                           "use build_from instead")
+
+        if build_from:
+            source_type, source_value = build_from.split(':', 1)
+            if source_type not in ('image', 'imagestream'):
+                raise OsbsValidationException(
+                    'first part in build_from, may be only image or imagestream')
+            if source_type == 'image':
+                self.build_image.value = source_value
+            else:
+                self.build_imagestream.value = source_value
+
+        self.platforms.value = platforms
+        self.platform.value = platform
+        self.koji_target.value = koji_target
+        self.koji_task_id.value = koji_task_id
+        self.filesystem_koji_task_id.value = filesystem_koji_task_id
+        self.koji_parent_build.value = koji_parent_build
+        self.flatpak.value = flatpak
+        self.flatpak_base_image.value = flatpak_base_image
+        self.isolated.value = isolated
+        self.scratch.value = scratch
+
+        if not flatpak:
+            if not base_image:
+                raise OsbsValidationException("base_image must be provided")
+            self.trigger_imagestreamtag.value = get_imagestreamtag_from_image(base_image)
+
+            if not name_label:
+                raise OsbsValidationException("name_label must be provided")
+            self.imagestream_name.value = name_label.replace('/', '-')
+
+        if signing_intent and compose_ids:
+            raise OsbsValidationException(
+                'Please only define signing_intent -OR- compose_ids, not both')
+        if compose_ids and yum_repourls:
+            raise OsbsValidationException(
+                'Please only define yum_repourls -OR- compose_ids, not both')
+        if not (compose_ids is None or isinstance(compose_ids, list)):
```

Comment:
Minor: can do this more flexibly with:
```
import collections
[...]
if isinstance(x, collections.Sequence):
```

(then it works for any sequence, e.g. tuples as well.

Reply:
strings are collections and so this fails an existing test case. Leaving it as it was.